README.md

```diff
@@ -60,6 +60,41 @@ need to upgrade:
 - Once you're happy it works, disable the old hubot integration from
   https://my.slack.com/services
 
+## Upgrading from version 3 or earlier of hubot-slack
+
+Version 4 of the hubot-slack adapter uses a more recent version of
+`node-slack-sdk`.  As a result, there are some syntax changes within Hubot:
+
+1. Before version 4, `msg.message.room` would return the name of the room
```

Comment:
It might be a good idea to mention that it also previously returned the slack username if the message was a direct message from a client.  I think the  following should work as a replacement `robot.adapter.client.rtm.dataStore.getUserById(room).name`